CMakeLists.txt

```diff
@@ -1,31 +1,64 @@
-cmake_minimum_required(VERSION 3.12)
+cmake_minimum_required(VERSION 3.14)
 
 include(pico_sdk_import.cmake)
 
 project(picoprobe)
 
+# On the next line, retain "pico" for Raspberry Pi Pico,
+# or set 'PICO_BOARD' to a board name as defined in
+# https://github.com/raspberrypi/pico-sdk/tree/master/src/boards/include/boards
+#
+# For example,
+#    "adafruit_qtpy_rp2040"    for the Adafruit QTPy
+#    "adafruit_feather_rp2040" for the Adafruit Feather RP2040
+#    "sparkfun_promicro"       for the SparkFun Pro Micro RP2040
+
+set(PICO_BOARD "pico")
+set(EXECUTABLE_NAME "picoprobe_${PICO_BOARD}")
 pico_sdk_init()
 
-add_executable(picoprobe
-        src/led.c
-        src/main.c
-        src/usb_descriptors.c
-        src/probe.c
-        src/cdc_uart.c
-        src/get_serial.c
+add_executable(${EXECUTABLE_NAME}
+               src/led.c
+               src/main.c
+               src/usb_descriptors.c
+               src/probe.c
+               src/cdc_uart.c
+               src/get_serial.c
 )
 
+if (PICO_BOARD STREQUAL "adafruit_qtpy_rp2040")
+    message("Building for Adafruit QTPy RP2040 with WS2812")
+elseif (PICO_BOARD STREQUAL "adafruit_feather_rp2040")
+    message("Building for Adafruit Feather RP2040 with WS2812")
+elseif (PICO_BOARD STREQUAL "sparkfun_promicro")
+    message("Building for SparkFun Pro Micro RP2040 with WS2812")
+else()
+    message("Building for Raspberry Pi Pico")
+endif()
+
+if (NOT PICO_BOARD STREQUAL "pico")
+    target_sources(${EXECUTABLE_NAME} PRIVATE src/ws2812.c)
```

Comment:
This seems to be assuming that any board which **isn't** a Raspberry Pi Pico will _always_ have a WS2812 LED, which I guess won't always be the case? Probably better to do something like `set(HAS_WS2812 1)` in the bit above where you're checking individual board-names?